### PR TITLE
Add allowed warnings to the dot-prefix yaml tests

### PR DESCRIPTION
**Pull request by dakrone**
This adds the missing optional warning from the global legacy template

Relates to #113529 (doesn't close it yet until this has been unmuted, which won't happen until this change is backported to 8.16)

### CODE REVIEW

File: modules/dot-prefix-validation/src/yamlRestTest/resources/rest-api-spec/test/dot_prefix/10_basic.yml [hunk 1]
```diff
@@ -177,7 +179,8 @@ teardown:
 
   - do:
       headers: { X-elastic-product-origin: kibana }
-      warnings:
+      allowed_warnings:
+        - "index template [my-template2] has index patterns [regular, .data-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-template2] will take precedence during new index creation"
```
Comment (nielsbauman) [inline modules/dot-prefix-validation/src/yamlRestTest/resources/rest-api-spec/test/dot_prefix/10_basic.yml:183]:
```suggestion
        - "index template [my-template2] has index patterns [other, .data2-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-template2] will take precedence during new index creation"
```

**Review by nielsbauman — APPROVED** [review]
LGTM - once you update the second index patterns list :)

File: modules/dot-prefix-validation/src/yamlRestTest/resources/rest-api-spec/test/dot_prefix/10_basic.yml (hunk 1 above)
Comment (dakrone) [reply modules/dot-prefix-validation/src/yamlRestTest/resources/rest-api-spec/test/dot_prefix/10_basic.yml:183]: Whoops, thanks!

**Comment by dakrone** [comment]
@elasticmachine update branch

**Pull request merged**